shunting-yard-common/src/main/java/com/hotels/shunting/yard/common/event/apiary/SerializableApiaryAddPartitionEvent.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.shunting.yard.common.event.apiary;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.events.AddPartitionEvent;
+
+import com.hotels.shunting.yard.common.event.SerializableListenerEvent;
+
+public class SerializableApiaryAddPartitionEvent extends SerializableListenerEvent {
+  private static final long serialVersionUID = 1L;
+
+  private String protocolVersion;
+  private String dbName;
+  private String tableName;
+  private Map<String, String> partitionKeys;
+  private List<String> partitionValues;
+
+  SerializableApiaryAddPartitionEvent() {}
+
+  public SerializableApiaryAddPartitionEvent(AddPartitionEvent event) {
+    super(event);
+    dbName = event.getTable().getDbName();
+    tableName = event.getTable().getTableName();
+    partitionKeys = new LinkedHashMap<>();
+    partitionValues = new ArrayList<>();
+
+    Iterator<Partition> iterator = event.getPartitionIterator();
+
+    // there will be only one partition in one event from Apiary
```

Comment:
Can we definitely be sure that only one partition will be in an event? Because `AddPartitionEvent` has a constructor that accepts a list of partitions too.
Also, if there will definitely be only one partition, there's no need to iterate through a partition in a while loop, right?

Reply:
Yeah. I thought about this but the thing here is that the only way to get a partition out from an event is through the iterator. So, I will anyway have to do a check for hasnext() and then a next() to get the first and only partition. Therefore, by having a loop here, actually handles both the if condition and the next().